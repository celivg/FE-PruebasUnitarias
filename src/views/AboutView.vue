<script>
export default {
  name: 'AboutView'
}
</script>

<template>
  <div>
    <h1>Acerca de</h1>
  </div>
</template>
