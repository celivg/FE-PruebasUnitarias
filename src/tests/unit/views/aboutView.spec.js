import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'
import { describe, it, expect, test } from 'vitest'

import App from '@/App.vue'
import AboutView from '@/views/AboutView.vue'

describe("Renderiza AboutView al navegar a la ruta", async () => {
  const routerPrueba = createRouter({
    history: createWebHistory(),
    routes: [
      {
        path: "/about",
        name: "about",
        component: AboutView
      }
    ]
  })

  it('Navega y renderiza la vista correspondiente', async () => {
    routerPrueba.push({ name: 'about' });

    await routerPrueba.isReady();

    const wrapper = mount(App, {
      global: {
        plugins: [routerPrueba]
      }
    });

    expect(wrapper.findComponent(AboutView).exists()).toBeTruthy();
  })

  test('La estructura HTML se mantiene estable', () => {
    const wrapper = mount(AboutView);

    expect(wrapper.html()).toMatchSnapshot();
  })
})
