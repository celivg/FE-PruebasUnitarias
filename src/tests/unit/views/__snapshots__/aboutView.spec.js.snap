// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renderiza AboutView al navegar a la ruta > La estructura HTML se mantiene estable 1`] = `
"<div>
  <h1>Acerca de</h1>
</div>"
`;
